src/budy/models/product.py

```diff
@@ -581,3 +592,51 @@ def _get_offset_offset(self, count, limit, kwargs):
         kwargs["id"] = {"$lt" : self.id}
         offset = cls.count(**kwargs)
         return offset
+
+    def _update_search_description(self):
+        tokens = []
+        self._add_tokens("short_description", tokens, pluralize = True)
+        self._add_tokens("product_id", tokens, pluralize = False)
+        self._add_tokens("brand.name", tokens, pluralize = True)
+        self._add_tokens("season.name", tokens, pluralize = True)
+        self._add_tokens("characteristics", tokens, pluralize = False)
+        self._add_tokens("colors.name", tokens, pluralize = True)
+        self._add_tokens("categories.name", tokens, pluralize = True)
+        self._add_tokens("collections.name", tokens, pluralize = True)
+        self._add_tokens("compositions.name", tokens, pluralize = True)
+        tokens_s = "|".join(tokens)
+        tokens_s = tokens_s.lower()
+        tokens_s = self._simplify(tokens_s)
+        self.search_description = tokens_s
+
+    def _add_tokens(self, attribute, tokens, pluralize = False):
+        attributes = attribute.split(".")
+        attribute = attributes[0]
+        attributes = attributes[1:] if len(attributes) > 1 else []
+        if not hasattr(self, attribute): return
+
+        value = getattr(self, attribute)
+        if not value: return
+
+        type_s = str(type(value))
+        values = value if type_s in ("<type 'list'>", "<type 'tuple'>", "<class 'appier.typesf._References'>") else [value]
+        for value in values:
+            for _attribute in attributes: value = getattr(value, _attribute)
+            tokens.append(value)
+            if pluralize: tokens.append(value)
+
+    def _simplify(self, value):
+        value = value.replace(u"á", "a")
+        value = value.replace(u"é", "e")
+        value = value.replace(u"í", "i")
+        value = value.replace(u"ó", "o")
+        value = value.replace(u"ú", "u")
+        value = value.replace(u"ã", "a")
+        value = value.replace(u"õ", "o")
+        value = value.replace(u"â", "a")
+        value = value.replace(u"ô", "o")
+        value = value.replace(u"ç", "c")
+        return value
+
+    def _pluralize(self, value):

```

Comment:
this should probably be in a appier string util and evolve to do better pluralization